contracts/src/v0.8/dev/transmission/4337/SCA.sol

```
@@ -0,0 +1,71 @@
+  error InvalidSignature(bytes32 operationHash, address owner);
+
+  // Assign the owner of this contract upon deployment.
+  constructor(address owner, address entryPoint) {
+    s_owner = owner;
+    s_entryPoint = entryPoint;
+  }
+
+  /// @dev Validates the user operation via a signature check.
+  /// TODO: Utilize a "validAfter" for a tx to be only valid _after_ a certain time.
+  function validateUserOp(
+    UserOperation calldata userOp,
+    bytes32 userOpHash,
+    uint256 /* missingAccountFunds - unused in favor of paymaster */
+  ) external returns (uint256 validationData) {
+    if (userOp.nonce != s_nonce) {
+      // Revert for non-signature errors.
+      revert IncorrectNonce(s_nonce, userOp.nonce);
+    }
+
+    // Verify signature on hash.
+    bytes32 fullHash = SCALibrary.getUserOpFullHash(userOpHash, address(this));
+    bytes memory signature = userOp.signature;
+    bytes32 r;
+    bytes32 s;
+    assembly {
+      r := mload(add(signature, 0x20))
+      s := mload(add(signature, 0x40))
+    }
+    uint8 v = uint8(signature[64]);
+    if (ecrecover(fullHash, v + 27, r, s) != s_owner) {
+      return _packValidationData(true, 0, 0); // signature error
+    }
+    s_nonce++;
+
+    // Unpack deadline, return successful signature.
+    (, , uint48 deadline, ) = abi.decode(userOp.callData[4:], (address, uint256, uint48, bytes));
+    return _packValidationData(false, deadline, 0);
+  }
+
+  /// @dev Execute a transaction on behalf of the owner. This function can only
+  /// @dev be called by the EntryPoint contract, and assumes that `validateUserOp` has succeeded.
+  function executeTransactionFromEntryPoint(address to, uint256 value, uint48 deadline, bytes calldata data) external {
+    if (msg.sender != s_entryPoint) {
+      revert NotAuthorized(msg.sender);
+    }
+    if (deadline != 0 && block.timestamp > deadline) {
+      revert TransactionExpired(deadline, block.timestamp);
+    }
+    to.call{value: value}(data);
```

should we handle errors here? I believe `call` can return `success==false` in which case we might fail silently

---

Adding a change to include some bubble-up logic.